iRecruit-ui/src/components/Profile/Profile.jsx

```diff
@@ -0,0 +1,175 @@
+  };
+
+  const handleLinkedinUrlChange = (event) => {
+    setLinkedinUrl(event.target.value);
+  };
+
+  const handleUpdate = (event) => {
+    event.preventDefault();
+    // save user profile: use localStorage or a state management library like Redux (Lee's suggestion)
+
+    // Create an object to store the profile information
+    const profileData = {
+      photo,
+      bio,
+      education, currentInstitution,
+      skills,
+      experience,
+      linkedinUrl
+    };
+
+    // Save the profile information to localStorage
+    localStorage.setItem("profile", JSON.stringify(profileData));
+
+    setProfileUpdated(true);
+
+    setTimeout(() => {
+      setProfileUpdated(false);
+    }, 3500);
+    console.log("Profile information saved successfully!");
+    // Display a success message to be sure it updated as needed.
+  };
+
+  return (
+    <div>
+      <h2>Edit Profile</h2>
+      <form onSubmit={handleUpdate}>
+        <div className="photo-container">
+          <div className="profile-photo-container">
+            {photo ? (
+              <img className="profile-photo" src={photo.dataURL} alt="Profile" />
+            ) : (
+              <div className="empty-photo">Empty</div>
+            )}
+          </div>
+          <label htmlFor="photo" className="photo-upload-label">
+            Upload Photo
+            <input
+              type="file"
+              id="photo"
+              accept="image/*"
+              onChange={handlePhotoChange}
+            />
+          </label>
+        </div>
+        <div>
+          <label htmlFor="bio">Bio:</label>
+          <textarea id="bio" value={bio} onChange={handleBioChange}></textarea>
+        </div>
+        <div>
+          <label htmlFor="education">Education:</label>
+          <input
+            type="text"
+            id="education"
+            value={education}
+            onChange={handleEducationChange}
+          />
+        </div>
+        <div>
+          <label htmlFor="currentInstitution">Current Institution:</label>
+          <input
+            type="text"
+            id="currentInstitution"
+            value={currentInstitution}
+            onChange={handleCurrentInstitutionChange}
+          />
+        </div>
+        <div>
+          <label htmlFor="skills">Skills:</label>
+          <input
+            type="text"
+            id="skills"
+            value={skills}
+            onChange={handleSkillsChange}
+          />
+        </div>
+        <div>
+          <label htmlFor="experience">Experience:</label>
+          <textarea
+            id="experience"
+            value={experience}
+            onChange={handleExperienceChange}
+          ></textarea>
+        </div>
+        <div>
+          <label htmlFor="linkedinUrl">LinkedIn URL:</label>
+          <input
+            type="text"
+            id="linkedinUrl"
+            value={linkedinUrl}
+            onChange={handleLinkedinUrlChange}
+          />
+        </div>
```

Comment:
This is very repetitive code, lets move this into a generator function i.e. something like:
`{renderInput(INPUT_TYPE.SKILLS, skills, handleSkillsChange)}`

Reply:
okay, I will see to using generator functions and clean the repeated codes